src/request/sign-transaction/SignTransaction.js

```diff
@@ -81,7 +81,8 @@ class SignTransaction {
             $feeSection.classList.remove('display-none');
         }
 
-        if (request.layout === SignTransactionApi.Layouts.CASHLINK) {
+        if (request.layout === SignTransactionApi.Layouts.CASHLINK
+         && Nimiq.BufferUtils.equals(transaction.data, Constants.CASHLINK_FUNDING_DATA)) {
             if (request.cashlinkMessage) {
```

Comment:
Why did you decide to split this up into a separate `if` again?

Reply:
Because according to UX designers, the "Funding cashlink" message should not be displayed, even when no cashlink message is set.